python/pyspark/sql/types.py

```diff
@@ -191,14 +191,25 @@ def needConversion(self):
 
     def toInternal(self, dt):
         if dt is not None:
-            seconds = (calendar.timegm(dt.utctimetuple()) if dt.tzinfo
-                       else time.mktime(dt.timetuple()))
+            seconds = 0.0
+            try:
+                seconds = (calendar.timegm(dt.utctimetuple()) if dt.tzinfo
+                           else time.mktime(dt.timetuple()))
+            except:
+                # On Windows, the current value is converted to a timestamp when the current value is less than 1970
+                seconds = (dt - datetime.datetime.fromtimestamp(int(time.localtime(0).tm_sec) / 1000)).total_seconds()
```

Comment:
Is this Windows specific issue?

Reply:
Yes, linux does not have this problem, and it should be a bug of python3, but this method can solve this problem.

Reply:
IIRC 1970 handling issue is not OS specific problem. It would be great if you link some reported issues related to that.